sdk/cosmos/azure_data_cosmos/src/handler/request_handler.rs

```diff
@@ -0,0 +1,142 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+use crate::cosmos_request::{AuthorizationTokenType, CosmosRequest};
+use crate::handler::retry_handler::{BackOffRetryHandler, RetryHandler};
+use crate::operation_context::OperationType;
+use crate::pipeline::CosmosPipeline;
+use crate::resource_context::{ResourceLink, ResourceType};
+use crate::{ItemOptions, PartitionKey};
+use azure_core::http::Response;
+
+/// Facade wrapping a `CosmosPipeline` to add retry/backoff when sending requests.
+///
+/// Builds a `CosmosRequest`, sets routing context, and dispatches through
+/// `BackOffRetryHandler` which applies exponential backoff for transient
+/// failures (e.g. throttling). Produces typed `Response<T>` while keeping
+/// the underlying pipeline pluggable.
+#[derive(Debug, Clone)]
+pub struct RequestHandler {
+    pipeline: CosmosPipeline,
+    retry_handler: BackOffRetryHandler,
+}
+
+impl RequestHandler {
+    /// Creates a new `RequestHandler` wrapping the provided `CosmosPipeline`.
+    ///
+    /// This is a lightweight constructor; the handler itself is stateless aside
+    /// from holding a cloneable reference to the underlying `CosmosPipeline`.
+    ///
+    /// # Arguments
+    /// * `pipeline` - An initialized `CosmosPipeline` used to send requests.
+    ///
+    /// # Examples
+    /// ```ignore
+    /// let handler = RequestHandler::new(pipeline.clone());
+    /// ```
+    pub fn new(pipeline: CosmosPipeline) -> Self {
+        Self {
+            pipeline,
+            retry_handler: BackOffRetryHandler,
+        }
+    }
+
+    pub async fn send<T>(
+        &self,
+        partition_key: PartitionKey,
+        body: Option<Vec<u8>>,
+        operation_type: OperationType,
+        resource_type: ResourceType,
+        options: Option<ItemOptions<'_>>,
+        resource_link: ResourceLink,
+    ) -> azure_core::Result<Response<T>> {
+        let mut cosmos_request = CosmosRequest::new(
+            operation_type,
+            resource_type,
+            None, // resource_id (RID) not yet known here
+            partition_key,
+            body,
+            false,
+            AuthorizationTokenType::Primary,
+            options,
+        );
+        cosmos_request.request_context.location_endpoint_to_route =
+            Option::from(resource_link.url(&self.pipeline.endpoint));
+
+        let item_options = cosmos_request.clone().options.unwrap_or_default();
+
+        let ctx = item_options
+            .method_options
+            .context
+            .with_value(resource_link.clone());
```

Comment:
We should only clone what we need to clone, which is the context in this case:


```suggestion
        let ctx = cosmos_request
            .options
            .method_options
            .context
            .clone()
            .with_value(resource_link.clone());
```

Reply:
As discussed, for simplicity we can remove the dependency on `request_handler` for now. We can add it back later if needed. 

Therefore this change is now no longer applies.